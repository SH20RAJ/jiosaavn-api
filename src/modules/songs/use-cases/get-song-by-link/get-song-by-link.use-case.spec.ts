import { beforeAll, describe, expect, test } from 'vitest'
import { GetSongByLinkUseCase } from '#modules/songs/use-cases'

describe('GetSongByLink', () => {
  let getSongByLink: GetSongByLinkUseCase

  beforeAll(() => {
    getSongByLink = new GetSongByLinkUseCase()
  })

  test('should return a song by link', async () => {
    const song = await getSongByLink.execute('OgwhbhtDRwM')

    expect(song[0]).toMatchSnapshot({
      playCount: expect.any(Number)
    })
  })
})
