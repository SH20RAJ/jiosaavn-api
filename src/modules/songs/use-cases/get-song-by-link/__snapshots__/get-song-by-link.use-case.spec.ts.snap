// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GetSongByLink > should return a song by link 1`] = `
{
  "album": {
    "id": "49599287",
    "name": "Houdini",
    "url": "https://www.jiosaavn.com/album/houdini/YKL-I3to3Oc_",
  },
  "artists": {
    "all": [
      {
        "id": "776593",
        "image": [],
        "name": "Caroline Ailin",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/caroline-ailin-songs/ksTT3ixnrvU_",
      },
      {
        "id": "1264841",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-500x500.jpg",
          },
        ],
        "name": "Danny L Harle",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/danny-l-harle-songs/8uc,o2eYVw8_",
      },
      {
        "id": "776593",
        "image": [],
        "name": "Caroline Ailin",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/caroline-ailin-songs/ksTT3ixnrvU_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "1264841",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/733/Super-Natural-Remixes-English-2016-500x500.jpg",
          },
        ],
        "name": "Danny L Harle",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/danny-l-harle-songs/8uc,o2eYVw8_",
      },
      {
        "id": "657585",
        "image": [],
        "name": "Kevin Parker",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/kevin-parker-songs/WZROEDcg-u8_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
      {
        "id": "1266339",
        "image": [],
        "name": "Tobias Jesso Jr.",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/tobias-jesso-jr.-songs/7n6h1Xoulxc_",
      },
      {
        "id": "657585",
        "image": [],
        "name": "Kevin Parker",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/kevin-parker-songs/WZROEDcg-u8_",
      },
      {
        "id": "1266339",
        "image": [],
        "name": "Tobias Jesso Jr.",
        "role": "music",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/tobias-jesso-jr.-songs/7n6h1Xoulxc_",
      },
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "singer",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
    "featured": [],
    "primary": [
      {
        "id": "1274170",
        "image": [
          {
            "quality": "50x50",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_50x50.jpg",
          },
          {
            "quality": "150x150",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_150x150.jpg",
          },
          {
            "quality": "500x500",
            "url": "https://c.saavncdn.com/artists/Dua_Lipa_004_20231120090922_500x500.jpg",
          },
        ],
        "name": "Dua Lipa",
        "role": "primary_artists",
        "type": "artist",
        "url": "https://www.jiosaavn.com/artist/dua-lipa-songs/r-OWIKgpX2I_",
      },
    ],
  },
  "copyright": "Under exclusive license to Warner Records UK, a division of Warner Music UK Limited, â„— 2023 Radical22 Limited",
  "downloadUrl": [
    {
      "quality": "12kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_12.mp4",
    },
    {
      "quality": "48kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_48.mp4",
    },
    {
      "quality": "96kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_96.mp4",
    },
    {
      "quality": "160kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_160.mp4",
    },
    {
      "quality": "320kbps",
      "url": "https://aac.saavncdn.com/091/0d3cfd6080a23f13ace1c38c3296e18f_320.mp4",
    },
  ],
  "duration": 185,
  "explicitContent": false,
  "hasLyrics": false,
  "id": "JdJ_osp0",
  "image": [
    {
      "quality": "50x50",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-50x50.jpg",
    },
    {
      "quality": "150x150",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-150x150.jpg",
    },
    {
      "quality": "500x500",
      "url": "https://c.saavncdn.com/091/Houdini-English-2023-20231110015054-500x500.jpg",
    },
  ],
  "label": "Warner Records",
  "language": "english",
  "lyricsId": undefined,
  "name": "Houdini",
  "playCount": Any<Number>,
  "releaseDate": "2023-11-09",
  "type": "song",
  "url": "https://www.jiosaavn.com/song/houdini/OgwhbhtDRwM",
  "year": 2023,
}
`;
